Add communication templates (#584)
opensearch-project/opensearch-build-libraries: resources/release/criteria-not-met-template.md

Hi ${RELEASE_OWNER}, </br>

> The below ${TYPE_OF_CRITERIA} criteria for your component has not been met. </br>

Reviewer: can there be multiple criteria that are not being met?

Author: Type is either entrance or exit criteria. Starting with this. Will keep on iterating on top of this to make it list, etc.

> Please review the issue and address it as soon as possible. </br>

Reviewer 2: ```suggestion
Please review the issue and address it with high priority. </br>
```
